discovery-provider/src/tasks/index_rewards_manager_backfill.py

```diff
@@ -737,42 +744,43 @@ def index_rewards_manager_backfill(self):
     # Define lock acquired boolean
     have_lock = False
     # Max duration of lock is 4hrs or 14400 seconds
-    update_lock = redis.lock("solana_rewards_manager_backfill_lock", timeout=14400)
+    update_lock = redis.lock("solana_rewards_manager_backfill_lock")
 
-    with db.scoped_session() as session:
-        stop_sig = (
-            session.query(IndexingCheckpoint.signature)
-            .filter(
-                IndexingCheckpoint.tablename == index_rewards_manager_backfill_tablename
+    try:
+        with db.scoped_session() as session:
+            stop_sig = (
+                session.query(IndexingCheckpoint.signature)
+                .filter(
+                    IndexingCheckpoint.tablename
+                    == index_rewards_manager_backfill_tablename
+                )
+                .first()
             )
-            .first()
-        )
-        if not stop_sig:
-            stop_sig = find_true_stop_sig(session, solana_client_manager, stop_sig)
             if not stop_sig:
+                stop_sig = find_true_stop_sig(session, solana_client_manager, stop_sig)
+                if not stop_sig:
+                    logger.info(
+                        "index_rewards_manager_backfill.py | Failed to find true stop signature"
+                    )
+                    return
                 logger.info(
-                    "index_rewards_manager_backfill.py | Failed to find true stop signature"
+                    f"index_rewards_manager_backfill.py | Found true stop_sig: {stop_sig}"
                 )
-                return
-            logger.info(
-                f"index_rewards_manager_backfill.py | Found true stop_sig: {stop_sig}"
-            )
-        else:
-            stop_sig = stop_sig[0]
+            else:
+                stop_sig = stop_sig[0]
 
-        if not stop_sig:
-            logger.info(
-                f"index_rewards_manager_backfill.py | No stop_sig found: {stop_sig}"
-            )
-            return
+            if not stop_sig:
+                logger.info(
+                    f"index_rewards_manager_backfill.py | No stop_sig found: {stop_sig}"
+                )
+                return
 
-        if check_if_backfilling_complete(session, solana_client_manager, redis):
-            logger.info(
-                "index_rewards_manager_backfill.py | Backfill indexing complete!"
-            )
-            return
+            if check_if_backfilling_complete(session, solana_client_manager, redis):
+                logger.info(
+                    "index_rewards_manager_backfill.py | Backfill indexing complete!"
+                )
+                return
```

Comment:
this is sneaky but I'm noticing that on the catch on L791, we raise again - if we do this, it means we won't hit the `finally` below and release the lock. maybe shouldn't re-raise?

Reply:
`finally` block should always run, even if we return here.

```
>>> def myFn():
...   try:
...     print('hello')
...     return
...   finally:
...     print('world')
... 
>>> myFn()
hello
world
>>> 
```

Reply:
```
>>> def fn():
...     try:
...             print("HI")
...             raise Exception("OH NO")
...     finally:
...             print("WORLD")
...
>>> fn()
HI
WORLD
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in fn
Exception: OH NO
```

Reply:
Haha yep, that one was a given - was responding to @piazzatron RE: return statements. I think this is fine, you can resolve

Reply:
Actually i reread what @piazzatron 's comment says. A more complete example:

```
>>> def myFn2():
...   try:
...     print('hello')
...     raise Exception('foo')
...   except:
...     raise Exception('bar')
...   finally:
...     print('world')
... 
>>> myFn2()
hello
world
Traceback (most recent call last):
  File "<stdin>", line 4, in myFn2
Exception: foo

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 6, in myFn2
Exception: bar
```